xapian-core/tests/harness/backendmanager_multi.cc

```diff
@@ -91,28 +113,44 @@ BackendManagerMulti::createdb_multi(const string& name,
     // Open NUMBER_OF_SUB_DBS databases and index files to them alternately so
     // a multi-db combining them contains the documents in the expected order.
     Xapian::WritableDatabase dbs;
-    const string& subtype = sub_manager->get_dbtype();
-    int flags = Xapian::DB_CREATE_OR_OVERWRITE;
-    if (subtype == "glass") {
-	flags |= Xapian::DB_BACKEND_GLASS;
-    } else {
-	string msg = "Unknown multidb subtype: ";
-	msg += subtype;
-	throw msg;
-    }
+
     string dbbase = db_path;
     dbbase += "___";
     size_t dbbase_len = dbbase.size();
-    string line = subtype;
-    line += ' ';
-    line += dbname;
-    line += "___";
+
     for (size_t n = 0; n < NUMBER_OF_SUB_DBS; ++n) {
-	dbbase += str(n);
-	dbs.add_database(Xapian::WritableDatabase(dbbase, flags));
+	const string &subtype = sub_managers[n]->get_dbtype();
+	int flags = Xapian::DB_CREATE_OR_OVERWRITE;
+	if (subtype == "glass") {
+	    flags |= Xapian::DB_BACKEND_GLASS;
+	    dbbase += str(n);
+	    dbs.add_database(Xapian::WritableDatabase(dbbase, flags));
+	    string line = subtype + " " + dbname + "___" + str(n);
+	    out << line << '\n';
```

Comment:
There's no reason to put the line to output in a string now since we rebuild it each time from scratch anyway - just write the elements to the `out` stream directly.